builder/azure/common/template/template.go

```diff
@@ -52,19 +60,19 @@ type OSDiskUnion struct {
 	Caching      compute.CachingTypes              `json:"caching,omitempty"`
 	CreateOption compute.DiskCreateOptionTypes     `json:"createOption,omitempty"`
 	DiskSizeGB   *int32                            `json:"diskSizeGB,omitempty"`
-	ManagedDisk  *compute.ManagedDiskParameters    `json:"managedDisk,omitempty"`
+	ManagedDisk  *ManagedDisk                      `json:"managedDisk,omitempty"`
 }
 
 type DataDiskUnion struct {
-	Lun          *int                           `json:"lun,omitempty"`
-	BlobURI      *string                        `json:"blobUri,omitempty"`
-	Name         *string                        `json:"name,omitempty"`
-	Vhd          *compute.VirtualHardDisk       `json:"vhd,omitempty"`
-	Image        *compute.VirtualHardDisk       `json:"image,omitempty"`
-	Caching      compute.CachingTypes           `json:"caching,omitempty"`
-	CreateOption compute.DiskCreateOptionTypes  `json:"createOption,omitempty"`
-	DiskSizeGB   *int32                         `json:"diskSizeGB,omitempty"`
-	ManagedDisk  *compute.ManagedDiskParameters `json:"managedDisk,omitempty"`
+	Lun          *int                          `json:"lun,omitempty"`
+	BlobURI      *string                       `json:"blobUri,omitempty"`
+	Name         *string                       `json:"name,omitempty"`
+	Vhd          *compute.VirtualHardDisk      `json:"vhd,omitempty"`
+	Image        *compute.VirtualHardDisk      `json:"image,omitempty"`
+	Caching      compute.CachingTypes          `json:"caching,omitempty"`
+	CreateOption compute.DiskCreateOptionTypes `json:"createOption,omitempty"`
+	DiskSizeGB   *int32                        `json:"diskSizeGB,omitempty"`
+	ManagedDisk  *ManagedDisk                  `json:"managedDisk,omitempty"`
```

Comment:
Am I right to assume this will be a breaking change? Is there a way we can avoid this potentially?

Reply:
I don't believe this is a breaking change, this object is our definition of the Azure VM template object, we define a structure of that object that we use before marshalling it into a JSON that is used to create the VM, adding this field just adds the structure to set the encryption ID.  If you don't set the encryption ID the managed_disk field the resulting JSON should be the same as it was before this change, since we were already defining the managedDIsk json object

Reply:
Oh yeah you're right, for some reason I assumed this was a change to the config for the builder, but it looks like it's completely internal, never mind!